.github/actions/custom-build/action.yaml

```diff
@@ -0,0 +1,16 @@
+name: Build
+description: "A reusable build action"
+
+inputs:
+  build:
+    required: true
+  shell:
+    required: true
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Build (Matrix override)
+      shell: ${{ inputs.shell }}
+      env: ${{ matrix.settings.env || fromJSON('{}') }}
+      run: ${{ inputs.build }}
```

Comment:
What is the advantage of extracting this when it is only used once and only has a single step? Otherwise, I would find it more straightforward to inline it again.